pkg/controller/vpcpeering/peering.go

```diff
@@ -393,37 +375,111 @@ func (e *external) Update(ctx context.Context, mg cpresource.Managed) (managed.E
+		}
+		if cr.Annotations == nil {
+			cr.Annotations = map[string]string{}
+		}
+		cr.Annotations[attributeModified] = "true"
+		err = e.kube.Update(ctx, cr)
+		if err != nil {
+			return managed.ExternalUpdate{}, awsclient.Wrap(err, "error update peering annotations")
+		}
 	}
 
-	_, err := e.client.ModifyVpcPeeringConnectionOptionsRequest(modifyVpcPeeringConnectionOptionsInput).Send(ctx)
-	if err != nil {
-		return managed.ExternalUpdate{}, errors.New(errModifyVpcPeering)
-	}
+	if !routeTableReady {
+		filter := ec2.Filter{
+			Name: aws.String("vpc-id"),
+			Values: []string{
+				*cr.Spec.ForProvider.VPCID,
+			},
+		}
+		describeRouteTablesInput := &ec2.DescribeRouteTablesInput{
+			Filters:    []ec2.Filter{filter},
+			MaxResults: aws.Int64(10),
+		}
+		routeTablesRes, err := e.client.DescribeRouteTablesRequest(describeRouteTablesInput).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, awsclient.Wrap(err, errDescribeRouteTable)
+		}
 
-	vpcAssociationAuthorizationInput := &route53.CreateVPCAssociationAuthorizationInput{
-		HostedZoneId: cr.Spec.ForProvider.HostZoneID,
-		VPC: &route53.VPC{
-			VPCId:     cr.Spec.ForProvider.PeerVPCID,
-			VPCRegion: route53.VPCRegion(*cr.Spec.ForProvider.PeerRegion),
-		},
+		for _, rt := range routeTablesRes.RouteTables {
+			createRouteInput := &ec2.CreateRouteInput{
+				RouteTableId:           rt.RouteTableId,
+				DestinationCidrBlock:   cr.Spec.ForProvider.PeerCIDR,
+				VpcPeeringConnectionId: cr.Status.AtProvider.VPCPeeringConnectionID,
+			}
+			createRouteRes, err := e.client.CreateRouteRequest(createRouteInput).Send(ctx)
+			if err != nil {
+				if aerr, ok := err.(awserr.Error); ok {
+					if aerr.Code() != "RouteAlreadyExists" {
+						return managed.ExternalUpdate{}, awsclient.Wrap(err, errCreate)
+					}
+				}
+			} else {
+				e.log.Info("Create route for route table", "RouteTableID", *rt.RouteTableId, "return", *createRouteRes.Return)
+			}
+		}
+		if cr.Annotations == nil {
+			cr.Annotations = map[string]string{}
+		}
+		cr.Annotations[routeTableEnsured] = "true"
+		err = e.kube.Update(ctx, cr)
+		if err != nil {
+			return managed.ExternalUpdate{}, awsclient.Wrap(err, "error update peering annotations")
+		}
 	}
-	_, err = e.route53Client.CreateVPCAssociationAuthorizationRequest(vpcAssociationAuthorizationInput).Send(ctx)
-	if aerr, ok := err.(awserr.Error); ok {
-		if strings.Contains(strings.ToLower(aerr.Code()), "already") {
-			return managed.ExternalUpdate{}, errors.New(errModifyVpcPeering)
+
+	if !hostZoneReady {
+		vpcAssociationAuthorizationInput := &route53.CreateVPCAssociationAuthorizationInput{
+			HostedZoneId: cr.Spec.ForProvider.HostZoneID,
+			VPC: &route53.VPC{
+				VPCId:     cr.Spec.ForProvider.PeerVPCID,
+				VPCRegion: route53.VPCRegion(*cr.Spec.ForProvider.PeerRegion),
+			},
+		}
+		_, err := e.route53Client.CreateVPCAssociationAuthorizationRequest(vpcAssociationAuthorizationInput).Send(ctx)
+		if err != nil && !isAlreadyCreated(err) {
```

Comment:
`isAlreadyCreated` function already checks error is nil.

Reply:
the condition is `error is not nil`, that is, a nil error is not `AlreadyCreated`, but we don't want to return a error if the error is nil